zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -1035,6 +1035,9 @@ private boolean inlineRemove(String id, boolean initiator) throws IOException {
     }
 
     File localRepoDir = new File(conf.getInterpreterLocalRepoPath() + "/" + id);
+    if (localRepoDir.getAbsolutePath().contains("..")) {
+      return removed;
+    }
     FileUtils.deleteDirectory(localRepoDir);
```

Comment:
@Reamer Because the following will be deleted whether the if statement is executed or not

Reply:
But is it logical to delete a localRepoDir without an interpreter existing?

Reply:
Normally it is unreasonable, maybe it can be changed to this?

Reply:
At first I would try to move the deletion to the if block. After that, a path check should be unnecessary.
If this solution doesn't work I would prefer such code.
```java
File localRepoDir = new File(conf.getInterpreterLocalRepoPath())
if (Arrays.stream(localRepoDir.list()).anyMatch(id::equals)) {
    FileUtils.deleteDirectory(new File(localRepoDir, id));
}
```
`..` is not a good solution, as this does not protect against attacks that delete files inside the `localRepoDir`.